File: zenox/auto_tasks/topGG.py
import os
import aiohttp
from typing import ClassVar
from sentry_sdk.crons import monitor
from zenox.bot.bot import Zenox
from zenox.static.utils import send_webhook

class TopGG:
    _bot: ClassVar[Zenox]
    _url = "https://top.gg/api/bots/781529450734551071/stats"

    @classmethod
    @monitor(monitor_slug='update-topgg-guild-count')
    async def execute(self, client: Zenox):
        self._bot = client
        if self._bot.env != "prod":
            return
        try:
            if token := os.getenv('TOPGG_TOKEN'):
                headers = {"Authorization": f"Bearer {token}"}
                data = {"server_count": len(client.guilds)}
                async with aiohttp.ClientSession() as sess:
                    async with sess.post(self._url, headers=headers, json=data) as response:
                        response.raise_for_status()
                        await send_webhook(client.log_webhook_url, username="TopGG Task", content="Updated Guild Count on TopGG")
        except Exception as e:
            self._bot.capture_exception(e)
            await send_webhook(client.log_webhook_url, username="TopGG Task", content="Failed to update Guild Count on TopGG")

File: src/auto_tasks/topGG.ts
import axios from 'axios';
import { Zenox } from '../bot/bot';
import { sendWebhook } from '../static/utils';
import { log } from '../utils/logger';

export class TopGG {
  private static bot: Zenox;
  private static readonly url = "https://top.gg/api/bots/781529450734551071/stats";

  static async execute(client: Zenox): Promise<void> {
    log.separator('TOP.GG UPDATE TASK');
    log.info('Starting Top.gg guild count update...');
    
    this.bot = client;
    
    if (this.bot.env !== "prod") {
      log.info('Skipping Top.gg update - not in production environment');
      return;
    }

    try {
      const token = process.env.TOPGG_TOKEN;
      if (!token) {
        log.warn('Top.gg token not found, skipping update');
        return;
      }

      const headers = { "Authorization": `Bearer ${token}` };
      const data = { "server_count": client.guilds.cache.size };

      log.info(`Updating Top.gg with ${data.server_count} guilds...`);

      const response = await axios.post(this.url, data, { headers });
      
      if (response.status === 200) {
        log.success('Successfully updated Top.gg guild count');
        
        await sendWebhook(
          client.logWebhookUrl!,
          "TopGG Task",
          "Updated Guild Count on TopGG"
        );
      } else {
        throw new Error(`Top.gg API returned status ${response.status}`);
      }
    } catch (error) {
      log.error('Failed to update Top.gg guild count', error);
      this.bot.captureException(error as Error);
      
      await sendWebhook(
        client.logWebhookUrl!,
        "TopGG Task",
        "Failed to update Guild Count on TopGG"
      );
    }

    log.separator('TOP.GG UPDATE COMPLETE');
  }
}